services/backend/src/components/profiles/profiles.hooks.js

```diff
@@ -67,12 +67,26 @@ const validatePersonalDetails = () => (context) => {
     }
 };
 
+const findOrCreateQueryCustomizer = (data) => {
+    let query = {};
+
+    if ("name" in data) {
```

Comment:
`id` should come first before `name`, otherwise sending a full profile object with both `id` and `name` will cause `name` to take precedence.